api/internal/tests/views/test_account_viewset.py

```diff
@@ -857,6 +857,111 @@ def test_update_must_fail_if_quantity_and_plan_are_equal_to_the_owners_current_o
+
+        for desired_plan in desired_plans:
+            response = self._update(
+                kwargs={
+                    "service": self.current_owner.service,
+                    "owner_username": self.current_owner.username,
+                },
+                data={"plan": desired_plan},
+            )
+
+            assert response.status_code == status.HTTP_400_BAD_REQUEST
+            assert response.json() == {
+                "plan": {
+                    "value": [
+                        f"Invalid value for plan: {desired_plan['value']}; must be one of ['users-basic', 'users-pr-inappm', 'users-pr-inappy']"
+                    ]
+                }
+            }
+
+    def test_update_team_plan_must_fail_if_too_many_activated_users_during_trial(self):
+        self.current_owner.plan = PlanName.BASIC_PLAN_NAME.value
+        self.current_owner.plan_user_count = 1
+        self.current_owner.trial_status = TrialStatus.ONGOING.value
+        self.current_owner.plan_activated_users = [i for i in range(11)]
+        self.current_owner.save()
+
+        desired_plans = [
+            {"value": PlanName.TEAM_MONTHLY.value, "quantity": 10},
+            {"value": PlanName.TEAM_YEARLY.value, "quantity": 10},
+        ]
+
+        for desired_plan in desired_plans:
+            response = self._update(
+                kwargs={
+                    "service": self.current_owner.service,
+                    "owner_username": self.current_owner.username,
+                },
+                data={"plan": desired_plan},
+            )
+
+            assert response.status_code == status.HTTP_400_BAD_REQUEST
+            assert response.json() == {
+                "plan": {
+                    "value": [
+                        f"Invalid value for plan: {desired_plan['value']}; must be one of ['users-basic', 'users-pr-inappm', 'users-pr-inappy']"
+                    ]
+                }
+            }
+
+    def test_update_team_plan_must_fail_if_currently_team_plan_add_too_many_users(self):
+        self.current_owner.plan = PlanName.TEAM_MONTHLY.value
+        self.current_owner.plan_user_count = 1
+        self.current_owner.save()
+
+        desired_plans = [
+            {"value": PlanName.TEAM_MONTHLY.value, "quantity": 11},
+            {"value": PlanName.TEAM_YEARLY.value, "quantity": 11},
+        ]
+
+        for desired_plan in desired_plans:
+            response = self._update(
+                kwargs={
+                    "service": self.current_owner.service,
+                    "owner_username": self.current_owner.username,
+                },
+                data={"plan": desired_plan},
+            )
+            assert response.status_code == status.HTTP_400_BAD_REQUEST
+            assert (
+                response.data["plan"]["non_field_errors"][0]
+                == "Quantity for Team plan cannot exceed 10"
+            )
+
+    def test_update_must_fail_if_team_plan_and_too_many_users(self):
```

Comment:
What's this test? Not sure I understand it

Reply:
If user is trying to upgrade to team plan and is requesting more than 10 seats.